pkg/util/acrtoken/acrtoken.go

```diff
@@ -98,14 +99,64 @@ func (m *manager) EnsureTokenAndPassword(ctx context.Context, rp *api.RegistryPr
 		return "", err
 	}
 
+	return m.generateTokenPassword(ctx, mgmtcontainerregistry.TokenPasswordNamePassword1, rp)
+}
+
+// RotateTokenPassword chooses either the unused token password or the token
+// password with the oldest creation date, generates a new password, and
+// then updates the registry profile with the newly generated password.
+func (m *manager) RotateTokenPassword(ctx context.Context, rp *api.RegistryProfile) error {
```

Comment:
Could we skip the rotation of password if it has `password.expiry` as null? Or will it add an undesired conditions in the function?

Reply:
We are currently not planning to set password expiration on the tokens, but still need to rotate them.